include/tachimawari/control/controller/platform/linux.hpp

```diff
@@ -40,10 +40,8 @@ class Linux
   void close_port();
   void clear_port();
 
-  int write_port(std::vector<uint8_t> packet);
-  int read_port(
-    std::shared_ptr<std::vector<uint8_t>> packet, int packet_length,
-    int packet_index);
+  int write_port(const std::vector<uint8_t> & packet);
+  int read_port(std::shared_ptr<std::vector<uint8_t>> packet, int packet_length, int packet_index);
```

Comment:
Can we pass `packet` by reference here?


```suggestion
  int read_port(std::shared_ptr<std::vector<uint8_t>> & packet, int packet_length, int packet_index);
```